hsm/CMakeLists.txt

```diff
@@ -0,0 +1,12 @@
+if(BUILD_SDF)
+    file(GLOB_RECURSE HEADERS "*.h")
+    file(GLOB_RECURSE SRC_LIST "*.cpp")
+    if(BUILD_STATIC)
```

Comment:
https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html

you only need this var and only call add_library once